### PR TITLE
Joblib trainable segmentation

### PR DESCRIPTION
**Pull request by emmanuelle**
Follow-up of #461 using joblib Memory instead of global dict as database.

### CODE REVIEW

**Comment by nicholas-esterer** [comment]
See my PR that fixes the included script and makes it so we don't need the "Save Classifier". If it's good then we can merge it to here :)
https://github.com/plotly/dash-sample-apps/pull/468

File: apps/dash-image-segmentation/app.py [hunk 1]
```diff
@@ -356,25 +368,45 @@ def show_segmentation(image_path, mask_shapes, segmenter_args):
...
+        ("Show segmentation" in show_segmentation_value)
+        and (len(masks_data["shapes"]) > 0)
+    ):
+        segmentation_features_dict = {
+            "intensity": False,
+            "edges": False,
+            "texture": False,
+        }
+        for feat in segmentation_features_value:
+            segmentation_features_dict[feat] = True
```
[the remark this answers is not in the document]
Comment (emmanuelle) [reply apps/dash-image-segmentation/app.py:393]: I don't think so, `segmentation_features_value` corresponds to the checked values (intensity, edges and texture are initialized to False above, and the for loop modifies only the ones which are checked by the user)

File: apps/dash-image-segmentation/trainable_segmentation.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+            number of dimensions compatible with ``self.features_func``.
+
+        Raises
+        ------
+        NotFittedError if ``self.clf`` has not been fitted yet (use ``self.fit``).
+        """
+        if self.features_func is None:
+            self.features_func = multiscale_basic_features
+        features = self.features_func(image)
+        return predict_segmenter(features, self.clf)
```
[the remark this answers is not in the document]
Comment (emmanuelle) [reply apps/dash-image-segmentation/trainable_segmentation.py:210]: yes, I copied the whole module submitted to scikit-image. You are right, no need to include here unused code, I will remove the unused parts.

**Comment by emmanuelle** [comment]
@nicholas-esterer  I merged your PR (thanks for it!), please tell me if it's ok to merge this one :-).

**Comment by emmanuelle** [comment]
I compared the deployment on playground and on the gallery and this branch is much faster :-).

File: apps/dash-image-segmentation/app.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicholas-esterer) [reply apps/dash-image-segmentation/app.py:393]: It's true, I must have gotten confused. Ok nvm :).

**Comment by emmanuelle** [comment]
I'll remove the unused code before merging.

**Comment by nicholas-esterer** [comment]
Yep, looks good! One little thing I noticed with many dash apps in general is their title in the browser tabs is always "Dash". Is there a way to change this to be more descriptive? But this is a small thing.
Good work! :woman_dancing:

**Comment by emmanuelle** [comment]
> Yep, looks good! One little thing I noticed with many dash apps in general is their title in the browser tabs is always "Dash". Is there a way to change this to be more descriptive? But this is a small thing.

Yes you can just set `app.title` as in 02bd8dd

**Pull request merged**